### DISCUSSION TITLE
Question about Steinhardt’s parameter q1

### DISCUSSION THREAD

**Discussion by Nyanaaa** [pyscal/pyscal#128] [Q&A] [answered]
Hi,

Congratulation and thanks for your package, its very nice and easy to use!

I have a question about the Steinhardt’s parameters. In your package, it is limited to q2 to q13, but in my case, I need q1 to q8.

So is there a way to estimate q1 with your package?

thanks by advance!

**Comment by srmnitc** [comment, marked as answer]
Thanks! This is possible with the new version of pyscal which is not yet complete. I can give you the instructions to use it here. First remove any existing version of pyscal (`pip uninstall pyscal`) or use a new conda environment. After that:

- Download the new version of pyscal [here](https://github.com/pyscal/pyscal/archive/refs/heads/v3.zip) and unzip it.
- Go to the folder and do `python setup.py install`
- Now you use use almost the same code as before, for example see [here](https://github.com/pyscal/pyscal/blob/v3/tests/test_q3.py#L14)

If you run into trouble, please let me know, I can help further.

**Reply by Nyanaaa** [reply]
Thank you! I will try this

**Reply by Nyanaaa** [reply]
I tried it actually, but I can only install package through pip on my lab cluster, so I had to install the new version on my computer. But my computer does not have enough RAM to measure the neighbors...

I have a dump file for a 20nm crystal (around 7000000 atoms, the dump.gz file is 70Mo), and the RAM consumption to estimate the neighbors is 35Gb with the function :
sys.find_neighbors(method='cutoff', cutoff='adaptive')

Is there a way to limit it?

**Reply by srmnitc** [reply]
Unfortunately not at the moment. This is one of the drawbacks of pyscal. The existing code does not allow for such huge systems. A workaround would be to somehow divide your system into smaller ones and try. I am still thinking about a possible solution, but no updates so far.
